### PR TITLE
pipeline: update actions

### PR DESCRIPTION
**Pull request by aarnq**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [x] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

**Comment by aarnq** [comment]
@crssnd please take a look and let me know what you think about the build step. If we add a similar one in e2e then we can build the cache there on the main branch so all other branches can access it.
Feels a bit less wasteful than the current method of rebuilding 850 MB worth of images in each job and we don't need to involve a registry for PRs.

**Comment by crssnd** [comment]
> @crssnd please take a look and let me know what you think about the build step. If we add a similar one in e2e then we can build the cache there on the main branch so all other branches can access it. Feels a bit less wasteful than the current method of rebuilding 850 MB worth of images in each job and we don't need to involve a registry for PRs.

yes, I think it's a good idea. we should try it

**Comment by aarnq** [comment]
I think I should test it for the end2end but I think I'll wait on the terraform upgrade to be merged first.

**Review by Xartos — COMMENTED** [review]
So do I understand correctly that this basically uses the cache as registry for the image? If so that's a very neat solution

**Comment by aarnq** [comment]
> So do I understand correctly that this basically uses the cache as registry for the image? If so that's a very neat solution

It does indeed. And caches are shared with subbranches so we should always have a fresh cache from the e2e.

**Review by Xartos — APPROVED** [review]

**Review by crssnd — APPROVED** [review]

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -22,42 +22,65 @@ jobs:
...
+      run: /usr/bin/docker run --rm -e "CK8S_CONFIG_PATH" -e "CK8S_CLOUD_PROVIDER" -e "CI_CK8S_FLAVOR" -e "CK8S_PGP_FP" -e "GITHUB_RUN_ID" -e CI=true -v "/home/runner/work/compliantkubernetes-apps/compliantkubernetes-apps":"/github/workspace" -w /github/workspace elastisys-ck8s-apps/app:latest bash -c "./pipeline/sops-pgp-setup.bash && ./pipeline/init-unit-tests.bash"
       env:
         CK8S_CLOUD_PROVIDER: ${{ matrix.cloud }}
         CI_CK8S_FLAVOR: ${{ matrix.flavor }}
         CK8S_PGP_FP: 529D964DE0BBD900C4A395DA09986C297F8B7757
     - name: Run linters
-      uses: ./pipeline
-      with:
-        args: bash -c "./pipeline/sops-pgp-setup.bash && ./pipeline/lint.bash"
+      run: /usr/bin/docker run --rm -e "CK8S_CONFIG_PATH" -e "CK8S_CLOUD_PROVIDER" -e "CI_CK8S_FLAVOR" -e "CK8S_PGP_FP" -e "GITHUB_RUN_ID" -e CI=true -v "/home/runner/work/compliantkubernetes-apps/compliantkubernetes-apps":"/github/workspace" -w /github/workspace elastisys-ck8s-apps/app:latest bash -c "./pipeline/sops-pgp-setup.bash && ./pipeline/lint.bash"
```
Comment (robinAwallace) [inline .github/workflows/pipeline.yml:44]: Those there exist a "uses" that takes a docker image?
My thought is just to clear up the one liner :smile:

Comment (robinAwallace) [reply .github/workflows/pipeline.yml:44]: Nvm did not really find any good option

Comment (aarnq) [reply .github/workflows/pipeline.yml:44]: Yeah, that is the problem with this approach.

**Pull request merged**